Login and sign up

flamesoft opened the pull request:
PR stories : 
- https://www.pivotaltracker.com/story/show/121676105 
- https://www.pivotaltracker.com/story/show/121675679

Activities:
- Displays login form
- Displays sign-up form
- Adds tests for login and sign-up
- Note: the omni auth hasn't been implemented yet
- Note: login with different roles is not implemented yet

Commit:
Add views for devise

Commit:
WIP- add tests for sign up
It needs to have more fields such as username

Commit:
Add username for sign up in devise parameter
WIP: test for sign up needs to have index page done.

Commit:
Merge branch 'development' into login

Commit:
Merge with the latest PR 13

Commit:
Fix test for registration
Move common test steps to the common_steps.rb

Commit:
Merge branch 'development' into login

Commit:
Add wrong password confirmation test
for sign up

Commit:
Add duplicate username test

Commit:
Add test for duplicated email is not allowed to register account

Commit:
Add test for blank password is not allowed for registration

Commit:
Add test - blank email is not allowed for registration

Commit:
Add test for login successfully

Commit:
Add test for failed login

Commit:
Change css styles for login and sign up pages

tochman commented on app/controllers/application_controller.rb:
```diff
@@ -2,8 +2,15 @@ class ApplicationController < ActionController::Base
   def current_user
     User.create(username: 'test', role: 'admin')
   end
+
```
I think we need to get rid of this hardcoded `current_user` method.

tochman replied:
Any decision on this?

tochman commented on app/views/devise/confirmations/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
We are using HAML in this project.

tochman commented:
Some comments:
- Tests looks good (will go over them i detail later) - great increase in coverage!
- Views needs to be written in HAML
- We are using Foundation and I wonder if the views are using the foundation classes?

flamesoft commented:
Thanks for the comments. I will make the changes tomorrow.

flamesoft commented:
The views are using the foundation classes because the classes are global. So no special changes need to make. I made only adjustment to the login and the signup views to center the positions of input fields.

tochman commented:
Okay, I see some classes like `actions` and `fields`, etc...
Another question, have you configured the application layout with flashes? I can´t find that.( to display errors)

Commit:
Change html.erb pages to use haml
Removed forgotten password choice
Removed unused code under mailer, unlock, and confirmation folders
because the functions are not used in the system

Commit:
Adjust style for login and sign up buttons

Commit:
Remove unused current_user method

tochman commented:
Okay thx. I see no issues with this PR now. Can you go over this @DanielitoAlbinsson @callea2 @diraulo ?

Commit:
Add test for invalid email

Commit:
Add unit test and validation method for email in User model

tochman commented on spec/models/user_spec.rb:
```diff
@@ -24,6 +24,11 @@
     it { is_expected.to validate_uniqueness_of(:username) }
     it { is_expected.to validate_presence_of(:role) }
+    context 'Validate email' do
+      it { should allow_value('user@example.com').for(:email) }
```
We are using `is_expected.to` or `expect` syntax in our specs. Plz change to that for consistency.

tochman commented on spec/models/user_spec.rb:
```diff
@@ -24,6 +24,11 @@
     it { is_expected.to validate_presence_of(:username) }
     it { is_expected.to validate_uniqueness_of(:username) }
     it { is_expected.to validate_presence_of(:role) }
+    context 'Validate email' do
```
spacing

Commit:
Adjust matcher and spaces

tochman commented:
In my opinion this one is ready to merge
